src/atoms/affine/HcatAtom.jl

```diff
@@ -29,41 +29,15 @@ monotonicity(x::HcatAtom) = ntuple(_ -> Nondecreasing(), length(x.children))
 
 curvature(::HcatAtom) = ConstVexity()
 
-evaluate(x::HcatAtom) = hcat(map(evaluate, x.children)...)
+evaluate(x::HcatAtom) = reduce(hcat, map(evaluate, x.children))
```

Comment:
To avoid the quadratic time naive version:
```suggestion
evaluate(x::HcatAtom) = reduce(hcat, collect(map(evaluate, x.children)))
```